src/atomate2/common/jobs/gruneisen.py

```diff
@@ -0,0 +1,250 @@
+        symmetry.append(sga.get_space_group_number())
+    set_symmetry = list(set(symmetry))
+    if len(set_symmetry) == 1:
+        jobs = []
+        phonon_yaml_dirs = dict.fromkeys(("ground", "plus", "minus"), None)
+        phonon_imaginary_modes = dict.fromkeys(("ground", "plus", "minus"), None)
+        for st in opt_struct:
+            # phonon run for all 3 optimized structures (ground state, expanded, shrunk)
+            phonon_kwargs = {}
+            if prev_calc_dir_argname is not None:
+                phonon_kwargs[prev_calc_dir_argname] = prev_dir_dict[st]
+            phonon_job = phonon_maker.make(structure=opt_struct[st], **phonon_kwargs)
+            phonon_job.append_name(f" {st}")
+            # change default phonopy.yaml file name to ensure workflow can be
+            # run without having to create folders, thus
+            # prevent overwriting and easier to identify yaml file belong
+            # to corresponding phonon run
+            phonon_job.jobs[-1].function_kwargs.update(
+                filename_phonopy_yaml=f"{st}_phonopy.yaml",
+                filename_band_yaml=f"{st}_phonon_band_structure.yaml",
+                filename_dos_yaml=f"{st}_phonon_dos.yaml",
+                filename_bs=f"{st}_phonon_band_structure.pdf",
+                filename_dos=f"{st}_phonon_dos.pdf",
+            )
+            jobs.append(phonon_job)
+            # store each phonon run task doc
+            phonon_yaml_dirs[st] = phonon_job.output.jobdirs.taskdoc_run_job_dir
+            phonon_imaginary_modes[st] = phonon_job.output.has_imaginary_modes
+
+        return Response(
+            replace=Flow(jobs),
+            output={
+                "phonon_yaml": phonon_yaml_dirs,
+                "imaginary_modes": phonon_imaginary_modes,
+            },
+        )
+    return Response(output={"space_groups": set_symmetry}, stop_jobflow=True)
+
+
+@job(
+    output_schema=GruneisenParameterDocument,
+    data=[GruneisenParameter, GruneisenPhononBandStructureSymmLine],
+)
+def compute_gruneisen_param(
+    code: str,
+    phonopy_yaml_paths_dict: dict,
+    phonon_imaginary_modes_info: dict,
+    kpath_scheme: str,
+    symprec: float,
+    mesh: tuple | float = (20, 20, 20),
+    structure: Structure = None,
+    **compute_gruneisen_param_kwargs,
+) -> GruneisenParameterDocument:
+    """
+    Compute Grueneisen parameters from phonon runs.
+
+    Requires phonopy yaml files from ground, expanded and contracted structures
+
+    phonopy_yaml_paths_dict: phonopy yaml files path for ground, expanded and
+        contracted structure phonon runs
+    phonon_imaginary_modes_info: dict with bool indicating if structure
+        has imaginary modes
+    mesh: sampling mesh, or kpoint density
+    structure: pymatgen structure object at ground state
+    compute_gruneisen_param_kwargs: kwargs for phonopy Grueneisen
+        api and pymatgen plotters
+    """
+    ground = phonopy.load(
```

Comment:
Does the phonon_yaml get stored in the task document? If so, is it better to use that as it will support distributing calculations over multiple compute resources?

Reply:
The whole yaml isn't getting stored. I will need to think about how to do it. While I agree with you about multiple computing resources, this adds data transfer costs. 

At the moment, the whole `prev_dir` concept also doesn't really agree with multiple computing resources, right?

Reply:
I think these files are too large to pass through the jobstore. Especially with remote submission, this will become really slow and nearly impossible to run on a larger scale. In the Lobster workflows, we also don't put the WAVECAR into the jobstore.

We should think about a safe way to handle file copying between different computing resources in the future instead.

Reply:
That's fine, thanks for checking.  @gpetretto has an open PR on this which I just need to have one final look over.